### PR TITLE
test_all_sandia: fix clang build list for kokkos-dev-2

### PR DESCRIPTION
**Pull request by ndellingwood**

### CODE REVIEW

**Comment by ndellingwood** [comment]
@jeffmiles63 the old script needs updating for testing on kokkos-dev-2. The clang build list error is not present in the cm_test_all_sandia script.

**Comment by ndellingwood** [comment]
Added updates for pgi testing on blake (Jenkins test also added) and for testing on mayer with arm.

File: scripts/testing_scripts/cm_test_all_sandia [hunk 1]
```diff
@@ -502,12 +502,12 @@ elif [ "$MACHINE" = "kokkos-dev-2" ]; then
 
   SKIP_HWLOC=True
 
-  BASE_MODULE_LIST="sems-env,sems-cmake/3.12.2,kokkos-env,kokkos-hwloc/1.10.1/base,sems-<COMPILER_NAME>/<COMPILER_VERSION>"
-  GCC91_MODULE_LIST="sems-env,sems-cmake/3.12.2,kokkos-env,kokkos-hwloc/1.10.1/base,<COMPILER_NAME>/<COMPILER_VERSION>"
-  NVCC_MODULE_LIST="sems-env,sems-cmake/3.12.2,kokkos-env,kokkos-hwloc/1.10.1/base,<COMPILER_NAME>/<COMPILER_VERSION>,sems-gcc/7.3.0"
+  BASE_MODULE_LIST="sems-env,kokkos-env,kokkos-hwloc/1.10.1/base,sems-<COMPILER_NAME>/<COMPILER_VERSION>"
```
Comment (jeffmiles63) [inline scripts/testing_scripts/cm_test_all_sandia]: I'm not sure that removing sems-cmake/3.12.2 works here.  The script does a module purge before loading the modules for the selected build, and the default cmake is 2.7 or something like that....Kokkos needs at least 3.10 for the cmake build process.

Comment (ndellingwood) [reply scripts/testing_scripts/cm_test_all_sandia]: @jeffmiles63 at line 500 sems-cmake/3.12.2 is loaded for all builds after the purge, so we won't have to worry about a version of cmake that is unsupported.

**Comment by ndellingwood** [comment]
This PR allows for expanded testing coverage on mayer (arm arches), kokkos-dev-2 (module list fixes to test clang and clang+cuda) and blake (pgi compiler version update), @jeffmiles63 comments were addressed above. @crtrott or @dalg24 could one of you merge?

**Pull request merged**